### PR TITLE
Refactored Slack image with svg icon in "Getting Started" Page

### LINKED ISSUE #5830: Refactor "Getting Started" page to replace slack-join-meeting-channel.svg with icon-slack.svg
### Overview
We need to store only a single Slack icon SVG file in the codebase, and use CSS to render the SVG in various colors throughout the website.  In this issue, we modify the "Getting Started" page, replacing an `img` element with a liquid include tag referencing `_includes/svg/icon-slack.svg` and modifying CSS as needed to maintain the current appearance. 

### Action Items
- [x] Open `pages/getting-started.html` in your IDE and identify the 'img' tag with src `assets/images/getting-started/proj-4.png`, and identify the rendered image in the current [Getting Started webpage](https://www.hackforla.org/getting-started)
- [x]  Replace the `img` tag identified in the previous step with a liquid tag `{% include 'filename' %}` that will reference the file `_includes/svg/icon-slack.svg`.  Note that in the liquid include tag syntax, `filename` is relative to the `_includes` folder, so `_includes` should not be part of the filename.  
- [x] The size and layout styles of the icon will require modification.  Use docker to view the local page and modify the CSS in `_sass\components/_getting-started-page.scss` as needed so that the appearance of the icon in the local page matches the current webpage. 

### Resources/Instructions
[Getting Started HTML](https://github.com/hackforla/website/blob/gh-pages/pages/getting-started.html)
[Getting Started webpage](https://www.hackforla.org/getting-started)
- This issue resulted from #5758

### PR DESCRIPTION
**Pull request by agutiernc**
Fixes #5830 

### What changes did you make?
  - Replaced the slack image icon with an svg file icon on steps 2, 4, and 5
  - Modified CSS to align the icon and make responsive

### Why did you make the changes (we will use this info to test)?
  - To update icon in svg format
  - To reduce redundancy in code on the "Getting Started" page
  - Make it simpler to change out the icon for a newer icon

### Screenshots of Proposed Changes Of The Website

<details>
<summary>Visuals before changes are applied</summary>

before-slack-icon-step2
before-slack-icon-step4-5

</details>

<details>
<summary>Visuals after changes are applied</summary>
  
after-slack-icon-step2
after-slack-icon-step4-5

</details>

### CODE REVIEW

**Comment by mshirk2** [comment]
Review ETA: 6pm 11/17/23
Availability: 10-6pm Weekdays

**Comment by jaasonw** [comment]
image
The linter error here is occurring because stylelint [no longer supports scss out of the box](https://stylelint.io/migration-guide/to-14/) and now needs to be extended with `stylelint-config-standard-scss`

**Review by roslynwythe — COMMENTED** [review]
Great work @agutiernc.  You have setup the PR branches correctly, and provided descriptive text and screenshots, and the page looks perfect in the browser.  Obviously the CSS is correct but it is a tiny bit confusing; for a few classes (for example, .step-img-icon-join-slack) there are multiple CSS rules that differ slightly in a few styles.  Could I ask what is the reasoning behind that choice?  Obviously it works fine in the browser; I just wonder if there is a way to accomplish the same goal in a manner that made it more clear which styles will be applied, for those of us who are not yet expert at CSS.

**Comment by agutiernc** [comment]
> Great work @agutiernc. You have setup the PR branches correctly, and provided descriptive text and screenshots, and the page looks perfect in the browser. Obviously the CSS is correct but it is a tiny bit confusing; for a few classes (for example, .step-img-icon-join-slack) there are multiple CSS rules that differ slightly in a few styles. Could I ask what is the reasoning behind that choice? Obviously it works fine in the browser; I just wonder if there is a way to accomplish the same goal in a manner that made it more clear which styles will be applied, for those of us who are not yet expert at CSS.

Thanks for the feedback @roslynwythe! I'm glad that you asked me about the CSS style I did for the Slack icons. Here's why I did it:

I noticed that three classes had the same properties, so I grouped them together to make the CSS code shorter. I also added a comment to indicate that those classes are for the Slack icon.

Then I grouped the other two classes that had 2 different properties from `.step-img-icon-cop`. They needed to have those properties to line up with the other icons and the list.

I understand that it might have been a bit hard to follow. I can change it back to separate classes if preferred.

**Comment by agutiernc** [comment]
> image The linter error here is occurring because stylelint [no longer supports scss out of the box](https://stylelint.io/migration-guide/to-14/) and now needs to be extended with `stylelint-config-standard-scss`

That explains it. Thanks for letting me know!

**Review by mshirk2 — DISMISSED** [review]
@agutiernc Well done with this pull request!

- The new branch is merging into the correct branch 'gh-pages'
- Original issue is linked
- PR description is clear and correct
- The specified change has been correctly implemented

**Comment by roslynwythe** [comment]
@jaasonw Thank you for your insight and the link to instructions to install and configure the SCSS extension.  The ER #5930 has been created and should be approved shortly.  If you wish to write the issue, please let me know.

**Comment by heejung-hong** [comment]
Availability for this week: 12pm - 6pm
My estimated ETA for completing this issue: 11/29

**Comment by roslynwythe** [comment]
@agutiernc I apologize for the delay in completing the review.  I felt that it was important to do testing in Safari but wasn't able to complete that testing so @heejung-hong kindly offered to do that in her review.  

Regarding the CSS rules, I tend to not want to create multiple rules with the same specificity because then the result depends on which rule appears first, which can create some confusion, and could be broken by someone reordering the rules in the file, but I understand that there are advantages to your approach as well, so I will approve this once cross-browser appearance is confirmed.

**Comment by heejung-hong** [comment]
Hello @agutiernc and @roslynwythe,
I've completed my testing in Safari and the Slack icon doesn't appear in the browser.
Please see screenshots below for reference.

Screenshot 2023-11-29 at 11 56 42 AM

Screenshot 2023-11-29 at 11 57 11 AM

Screenshot 2023-11-29 at 11 57 36 AM

**Comment by agutiernc** [comment]
> Hello @agutiernc and @roslynwythe, I've completed my testing in Safari and the Slack icon doesn't appear in the browser. Please see screenshots below for reference.
> 

Thanks for confirming @heejung-hong. I think I figured out the issue with Safari. Safari's WebKit isn't up to date with SVG specifications yet. I'll have a fix in a few moments that I've been trying out in different browsers, including Safari.

**Comment by agutiernc** [comment]
> @agutiernc I apologize for the delay in completing the review. I felt that it was important to do testing in Safari but wasn't able to complete that testing so @heejung-hong kindly offered to do that in her review.
> 
> Regarding the CSS rules, I tend to not want to create multiple rules with the same specificity because then the result depends on which rule appears first, which can create some confusion, and could be broken by someone reordering the rules in the file, but I understand that there are advantages to your approach as well, so I will approve this once cross-browser appearance is confirmed.

@roslynwythe Thanks for having it tested in Safari. I wasn't aware that Safari's webkit doesn't render SVG properly. I learned something new today and implemented a fix.

In regards to the CSS rules, I'll keep a note about it for future issues.

**Comment by ajb176** [comment]
ETA: Sunday night if requires <3 hours, otherwise 12/9

**Review by ajb176 — CHANGES_REQUESTED** [review]
The page looks good to me, the Slack icon seems to be stable between Firefox and Safari. Nice work navigating through an issue that ended up a little more complex than advertised @agutiernc 

The class definitions are a little confusing, specifically the fact that the join and choose classes have their margin properties defined twice. If the goal is to make the code more concise, it's more intuitive to have only the common properties grouped together and defining the variant properties separately, meaning omitting the margins from the common definition altogether and defining it separately for the cop class as you did for the grouped join and choose classes. Maintaining the modularity from before can also make sense because the link between the choose and join classes seems a little tenuous (it seems to just be that they're both the topmost icons in their boxes) and may not always hold.

I'd suggest choosing one route or the other, but you've fixed the cross-browser issue which was the main concern so you can re-request and I'll approve if you'd rather keep things as is. Either way, nice work fixing the SVG.

**Review by heejung-hong — APPROVED** [review]
Good work!
The branches are correct and the issue is linked.
The correct file was changed with the correct information.
I have inspected and confirmed the changes in developer mode Chrome, Firefox, and Safari and the Slack icons show up on all three browsers.

**Comment by LRenDO** [comment]
Hi @agutiernc! Thanks for working on this issue and it's revisions! I am inclined to agree with @roslynwythe and @ajb176 about the CSS rules. While not crucial for functionality, it would help clarity and make it easier for folks making changes in the future if it was revised to one of the two options @ajb176 suggested. Please either request a re-review from @ajb176 if you would like to keep the CSS as-is or provide an ETA if you plan to revise it. Thanks!

**Comment by agutiernc** [comment]
> Hi @agutiernc! Thanks for working on this issue and it's revisions! I am inclined to agree with @roslynwythe and @ajb176 about the CSS rules. While not crucial for functionality, it would help clarity and make it easier for folks making changes in the future if it was revised to one of the two options @ajb176 suggested. Please either request a re-review from @ajb176 if you would like to keep the CSS as-is or provide an ETA if you plan to revise it. Thanks!

Hi @LRenDO, I'll revise the CSS rules and re-request a review.

ETA: EoD today - 1/10

**Comment by agutiernc** [comment]
I am closing this pull request because I accidentally merged the upstream changes from the original repository into my feature branch, and pushed them to my forked repository. This made the pull request very messy and hard to review. I apologize for the inconvenience and confusion this may have caused.

I will create a new pull request with the updated CSS changes.

Thank you all for understanding and patience.
